unit_test/test_acknowledgements.py

```diff
@@ -0,0 +1,275 @@
+            self.assertEqual("", lines[11])
+            self.assertEqual(self.LAST_LINE, lines[12])
+
+    def test_contents_preserved(self):
+        '''
+        Test case for the situation where the output file has contents that
+        have to be preserved.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.write_text("Line One.\nLine 2,\nline three.\n")
+            language = "nl-NL"
+            volunteers = ["Volunteer One"]
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertIsNotNone(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(9, len(lines))
+            self.assertEqual("Line One.", lines[0])
+            self.assertEqual("Line 2,", lines[1])
+            self.assertEqual("line three.", lines[2])
+            self.assertEqual("***", lines[3])
+            self.assertEqual(self.FIRST_LINE, lines[4])
+            self.assertEqual("", lines[5])
+            self.assertEqual("Volunteer One", lines[6])
+            self.assertEqual("", lines[7])
+            self.assertEqual(self.LAST_LINE, lines[8])
+
+    def test_trailing_blanks_removed(self):
+        '''
+        Test case for the situation where the output file has contents that
+        have to be preserved, except for trailing blanks.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.write_text("This is the end.\n\n    \n\n\t\n\n")
+            language = "nl-NL"
+            volunteers = ["Volunteer One"]
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertIsNotNone(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(7, len(lines))
+            self.assertEqual("This is the end.", lines[0])
```

Comment:
Same here. Comparing to a string would be more readable.

Reply:
Agreed; simplified